enhancements/network/core-network-policies.md

```diff
@@ -0,0 +1,532 @@
+
+
+### Implementation Details/Notes/Constraints
+
+What are some important details that didn't come across above in the
+**Proposal**? Go in to as much detail as necessary here. This might be
+a good place to talk about core concepts and how they relate. While it
+is useful to go into the details of the code changes required, it is
+not necessary to show how the code will be rewritten in the
+enhancement.
+
+
+#### DNS
+
+Almost every pod will need to have an egress policy to allow DNS.
+Would it be better to have one policy that can be enabled by a label?
+Or just have an admin network policy that just allows it.
+
+
+#### Host Network Pods
+
+Network Policy does not apply to pods that are running in the host
+network.  However, other pods that expect to receive connections from
+the host network on nodes will need
+[policy](https://docs.openshift.com/container-platform/4.17/networking/network_security/network_policy/about-network-policy.html)
+to allow it.
+
+
+#### API Server
+
+The API servers have to be specified by IP address, labels do not
+work.  We need a policy that is maintained by an operator that has a
+well-known label that pods can have applied to them.
+
+
+#### Monitoring
+
+
+#### Logging
+
+
+#### Operators
+
+Open issue: Who manages the operator namespaces?  They obviously need
+API server access to make the other objects, but who defines them.
+
+
+### Risks and Mitigations
+
+
+#### Policies Too Restrictive
+
+If we define policies that are too restrictive, valid flows can be
+blocked.  In some cases, the product allows for user-configured
+endpoint addresses, so we could have missed that in product testing.
+
+When this issue is identified, it can quickly be tested by using an
+Admin Network Policy on the namespace (or globally) to temporarily
+allow all traffic.  If that mitigates the issue, then a more targeted
+Admin Network Policy can be used to override our policies until a fix
+can be delivered in a z-stream.
+
+
+#### Policies Too Open
+
+The converse can also happen… we may write a policy that a customer
+finds is too open, or they may wish to impose additional restrictions
+that we may allow in general (for instance, something could allow a
+webhook to be called off-cluster, but they may want to forbid all
+off-cluster traffic from the namespace since they do not use that
+feature).
+
+Customers can write admin network policies that override the network
+policies that we ship.
+
+
+#### Challenges Debugging and Supporting
+
+When a network policy blocks traffic, it may be blocking traffic of an
+internal service communication causing a mysterious failure message to
+the user.
+
+You can use the Network Observability tool to audit network policy
+blocks.  There should not be any in the OpenShift namespaces.
+
+We also need to provide this information to the support organization
+in a must-gather.  We will augment the must-gather tool to include
+information about network policy blocks in OpenShift namespaces.
```

Comment:
Note that sufficiently-broken policies could interfere with both the Network Observability tool and must-gather.

Reply:
Hopefully we can catch at least some of that in automated testing.  I am assuming that these are the RH shipped policies, not random ones that a customer applied to us.